### ISSUE TITLE
View Complainant's Name and Contact information on the View All Table

### ISSUE THREAD

**Issue by alexsoble** [18F/crt-portal#26]
## Narrative
As an authenticated DOJ CRT user, 
I want to view a complainant's contact information
so that I can follow up with that complainant to
- request further information about their case
- communicate their case outcome to them

## Background
Public users can choose to give their contact information to CRT if they are submitting a complaint for themselves or someone else. They can also opt to report anonymously. Complainant contact information allows CRT sections to follow up with that individual to gather more information about their complaint. It does *not* prevent them from taking action on a case. If a complainant reports anonymously, CRT will not be able to contact them.

CRT users want to view complainants Name and Contact information on the View All table so that they can also spot frequent submitters.

## In scope
- manual sign-on page to access the admin portal

## Out of scope
- generating Unique ID of submitted complaint
- Filtering on table
- Sorting on table
- Internal CRT permissions
- viewing count for the ability to use daily/monthly/yearly data to produce reports.  
- styling on table — focus on functionality/content only

## Assumptions
- user has been authenticated via CRT SSO and can view this page
- we should also be able to capture 'Submission date' metadata

##### Given (this pre context)..., When (action happens)..., Then (result)...
1)
Given a violation has been reported by a complainant, 
When I log on to the admin portal
Then I see a landing page titled 'View All Complaint Records'
AND
Below the title I see a table with the following columns
- Contact name
- Contact details
- Submitted

2)
Given the complainant's name is John (First) Doe (Last
When I am viewing the Complainant name column
Then I view the complainant's name of the submitted record in the following format:

Doe, John 

3)
Given John Doe's email address is john@doe.com
AND his phone number is +270763208 (International ZA)
When I am viewing the Contact name column
Then I view the complainant's contact information of the submitted record in the following format:
john@doe.com
+270763208

3)
Given John Doe submitted his complaint on September 30, 2019 at 11:15 AM
When I am viewing the Contact details column
Then I view the complainant's contact information of the submitted record in the following format:

09/30/2019 (MM/DD/YYYY)

2)
Given CRT has received no submitted complaints
When I am viewing the 'View all complaint records' table
Then I view the following message
"There are no complaint records at this time" (UX to update text, if necessary)

## UX
![ViewAll_Default post-section feedback.png](https://images.zenhubusercontent.com/5d1a3b0bcc794602e102a513/b8f8fabf-1e06-49ef-9ddc-ac5c339ef88a)
screenshot of updates after section feedback
- Numeric date
- First name, Last name format
- Contact is three lines: Email, Number, City & State

[Expected UX](https://docs.google.com/presentation/d/1k0HWEhcIb6QC5fXNaXmePqoFpKau6_d-mZL2Q4DCa-w/edit#slide=id.g611277aa76_0_172)

**Comment by Jacklynn** [comment]
Lightly started this, hope to have something to show early next week.

**Comment by Jacklynn** [comment]
@blaughman is there a specific reason we *should* truncate the first name rather than showing it if there's room?

**Comment by Jacklynn** [comment]
Added current draft of expected UX to the summary above. 
https://docs.google.com/presentation/d/1k0HWEhcIb6QC5fXNaXmePqoFpKau6_d-mZL2Q4DCa-w/edit#slide=id.g611277aa76_0_172

**Comment by Jacklynn** [comment]
Updated design more based on feedback and more explorations: https://docs.google.com/presentation/d/1k0HWEhcIb6QC5fXNaXmePqoFpKau6_d-mZL2Q4DCa-w/edit?ts=5d769e0b#slide=id.g6099c81087_0_1
@blaughman @UchMok

**Comment by Jacklynn** [comment]
Q: If someone leaves a field blank, is there something we should call it to make to show fields that might be blank too? Like if someone fills something out anonymously?

**Comment by LindsayYoung** [comment]
The default for empty data is "None"

**Comment by Jacklynn** [comment]
Team feedback - 
- Text: Complainant name/contact > "contact" name/contact
- Text: Submitted (instead of submission date)
- Email and phone fixed to one line, no wrapping
- If blank field, "Blank" italicized so can tell if someone wrote nothing vs wrote nothing. Suggesting blank so that we can include a filter sort "include blank" type thing. 
- Title change to - "CRT Complaint Records"

- Remember to check in IE 11!

**Comment by LindsayYoung** [comment]
Adding the IE elements to https://github.com/18F/crt-portal/issues/138

**Comment by LindsayYoung** [comment]
Ok I think this should be ready to go, with the caveat that the contact page PR needs to be merged first since that is where the test fixes and the IE fixes are 
https://github.com/usdoj-crt/crt-portal/pull/53

**Comment by Jacklynn** [comment]
Contact --> Contact details

Also updated the AC

**Comment by UchMok** [comment]
@LindsayYoung @alexsoble how do we test the story? What's the URL?

**Comment by LindsayYoung** [comment]
We will make some accounts for the team this morning- the url will be https://crt-portal-django-dev.app.cloud.gov/form/view

**Comment by alexsoble** [comment]
@UchMok One quick note — Lindsay and I talked with the designers yesterday and agreed that styling is off the table for this Story. We'll focus on functionality only for this Story, and add in styling as we continue to build out the table. I'll make a note of that in the AC.

**Comment by Jacklynn** [comment]
Notes from review: 
* Remove "All reports view" title. Might be distracting to have this.

**Comment by UchMok** [comment]
@LindsayYoung @alexsoble remove text 'All Reports View'

**Comment by LindsayYoung** [comment]
"All Reports View" has been removed and auto-deployed to dev

**Comment by blaughman** [comment]
The View All Table works as it should in IE11

**Comment by lalitha-jonnalagadda** [comment]
We should be able to provide logins for the team to UAT the intake process.

**Comment by lalitha-jonnalagadda** [comment]
This functionality is deployed in staging and prod. Good to close this ticket.
